### ISSUE TITLE
Compile issue with boost in Ubuntu 18.04 and 16.04....

### ISSUE THREAD

**Issue by Romeo01234** [vergecurrency/verge#1041]
Hello Justin and Marpe, I should have been a little more clear on the previous post, issue #1040. Working nights and all.

So I have boost version 1.67 - 1.73 installed on both machines and i direct the "./configure" to the appropriate boost version, as well as openssl-1.1.1g.

I get the same compile error complaining about "boost.......deprecated..." on both machines of different OSes(18.04 and 16.04).

What boost version are you using to compile? I'll just install that one to see if I can get it working.

I may not respond right away cause of the night shift work. So let me know of what possible solutions you have for the boost problem even on the "Ubuntu 18.04".

Thanks and keep the good work going.

**Comment by marpme** [comment]
Could you maybe add the specific commands you used and what the log showed you. Otherwise I would have to blindly guess what's wrong there.

**Comment by Romeo01234** [comment]
Hello Marpme, my config setting are below:

./configure LDFLAGS="-L${BDB_PREFIX}/lib/" CPPFLAGS="-I${BDB_PREFIX}/include/" --enable-hardening --with-boost-libdir=/home/joe/boost_1_73_0/stage/lib --with-boost=/home/joe/boost_1_73_0/ LDFLAGS="-L/home/joe/openssl-1.1.1g/" CPPFLAGS="-I/home/joe/openssl-1.1.1g/include/openssl/"

**Comment by marpme** [comment]
Could you try to use your local openssl version that is coming with your current OS since the boost linkage looks fine and I remember there are incompatibilities with latest openssl

**Comment by Romeo01234** [comment]
I have tried these versions of openssl-1.1.1, 1.1.d, 1.1.1g. That is why i'm stumped. I checked to see what was the original openssl version for 18.04, I found a reference to the openssl version being v1.1.1. I had tried that and the make fails.

If it is different version, let me know the openssl version and i'll give it a try.  Better yet, what does the version you use say? 
I'll try that.

Thanks.

**Comment by Romeo01234** [comment]
I have tried the following './configure' below removing the OpenSSL, the error I get complains about the SSL.

./configure LDFLAGS="-L${BDB_PREFIX}/lib/" CPPFLAGS="-I${BDB_PREFIX}/include/" --enable-hardening --with-boost-libdir=/home/lazarus/boost_1_73_0/stage/lib --with-boost=/home/lazarus/boost_1_73_0/



In file included from src/lib/tls/tortls_openssl.c:61:0:
./src/lib/tls/tortls_internal.h:55:8: error: conflicting types for ‘SSL_SESSION_get_master_key’
 size_t SSL_SESSION_get_master_key(struct ssl_session_st *s,
        ^
In file included from src/lib/tls/tortls_openssl.c:48:0:
/usr/local/include/openssl/ssl.h:2007:15: note: previous declaration of ‘SSL_SESSION_get_master_key’ was here
 __owur size_t SSL_SESSION_get_master_key(const SSL_SESSION *sess,
               ^
src/lib/tls/tortls_openssl.c: In function ‘tor_tls_client_is_using_v2_ciphers’:
src/lib/tls/tortls_openssl.c:931:20: error: dereferencing pointer to incomplete type ‘SSL_SESSION {aka struct ssl_session_st}’
   ciphers = session->ciphers;
                    ^
src/lib/tls/tortls_openssl.c: At top level:
src/lib/tls/tortls_openssl.c:1579:1: error: conflicting types for ‘SSL_get_client_random’
 SSL_get_client_random(SSL *s, uint8_t *out, size_t len)
 ^
In file included from src/lib/tls/tortls_openssl.c:48:0:
/usr/local/include/openssl/ssl.h:2003:15: note: previous declaration of ‘SSL_get_client_random’ was here
 __owur size_t SSL_get_client_random(const SSL *ssl, unsigned char *out,
               ^
In file included from ./src/lib/crypt_ops/crypto_rand.h:16:0,
                 from src/lib/tls/tortls_openssl.c:31:
src/lib/tls/tortls_openssl.c: In function ‘SSL_get_client_random’:
src/lib/tls/tortls_openssl.c:1584:15: error: dereferencing pointer to incomplete type ‘SSL {aka struct ssl_st}’
   tor_assert(s->s3);
               ^
./src/lib/cc/compat_compiler.h:145:49: note: in definition of macro ‘PREDICT_LIKELY’
 #define PREDICT_LIKELY(exp) __builtin_expect(!!(exp), 1)
                                                 ^
./src/lib/log/util_bug.h:105:7: note: in expansion of macro ‘ASSERT_PREDICT_LIKELY_’
   if (ASSERT_PREDICT_LIKELY_(expr)) {                                   \
       ^
./src/lib/log/util_bug.h:102:26: note: in expansion of macro ‘tor_assertf’
 #define tor_assert(expr) tor_assertf(expr, NULL)
                          ^
src/lib/tls/tortls_openssl.c:1584:3: note: in expansion of macro ‘tor_assert’
   tor_assert(s->s3);
   ^
src/lib/tls/tortls_openssl.c: At top level:
src/lib/tls/tortls_openssl.c:1592:1: error: conflicting types for ‘SSL_get_server_random’
 SSL_get_server_random(SSL *s, uint8_t *out, size_t len)
 ^
In file included from src/lib/tls/tortls_openssl.c:48:0:
/usr/local/include/openssl/ssl.h:2005:15: note: previous declaration of ‘SSL_get_server_random’ was here
 __owur size_t SSL_get_server_random(const SSL *ssl, unsigned char *out,
               ^
src/lib/tls/tortls_openssl.c:1605:1: error: conflicting types for ‘SSL_SESSION_get_master_key’
 SSL_SESSION_get_master_key(SSL_SESSION *s, uint8_t *out, size_t len)
 ^
In file included from src/lib/tls/tortls_openssl.c:48:0:
/usr/local/include/openssl/ssl.h:2007:15: note: previous declaration of ‘SSL_SESSION_get_master_key’ was here
 __owur size_t SSL_SESSION_get_master_key(const SSL_SESSION *sess,
               ^
Makefile:16641: recipe for target 'src/lib/tls/src_lib_libtor_tls_a-tortls_openssl.o' failed
make[2]: *** [src/lib/tls/src_lib_libtor_tls_a-tortls_openssl.o] Error 1
make[2]: Leaving directory '/home/joe/coinz/verge-6.0.2Stealth_4.3.5/src/tor'
Makefile:6091: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/joe/coinz/verge-6.0.2Stealth_4.3.5/src/tor'
Makefile:786: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1

**Comment by Romeo01234** [comment]
The next step is for me to try and compile the code under "Virtualbox" only updating and Install the boost version(1.73) and see what the results are.

I'll do my best to get it done quickly to let you know if it passes or fails.

Thanks for patience.

**Comment by justinvforvendetta** [comment]
on a fresh distribution, you will have no problems if you:

```
sudo apt-get update && sudo apt-get install software-properties-common && sudo add-apt-repository ppa:bitcoin/bitcoin

sudo apt-get update && sudo apt-get install libdb4.8-dev libdb4.8++-dev build-essential libtool autotools-dev automake pkg-config bsdmainutils python3 libssl-dev libevent-dev libboost-all-dev zlib1g-dev libseccomp-dev libcap-dev libncap-dev && git clone https://github.com/vergecurrency/VERGE.git && cd VERGE && ./autogen.sh && ./configure --disable-bench --disable-tests && make

```

**Comment by Romeo01234** [comment]
Hello Justin and Marpme, the following was discovered during a recompile of updated Verge with Tor-4.3.5.

So the verge(with tor-4.3.5) compiled fine under VirtualBox with out of the compilation. This yielded Boost-1.65 and openssl-1.1.1.

I had tried to compile verge(with tor-4.3.5) under VirtualBox with Boost-1.73.0. It seems there is an issue with the Boost-1.73.0.
when I run the ./configure it complains about boost::sleep.
OpenSSL-1.1.1g works fine under compilation in VirtualBox.

I had tried to compile Boost-1.73.0 with the following various settings listed below, there are other ./configure settings I could try and will later. (Keep in mind I had also tried to compile boost with Python-3.8.3).

tried the following for _**boost-1.73.0:

./configure --with-libraries=all --with-python=/home/joe/Python-3.8.3/python
--->> Compiles fine some complaints about the "Deprecation", fails for Verge(tor-4.3.5).

./configure --with-libraries=all --with-python=python3 (System version 3.6)
--->> Compiles fine some complaints about the "Deprecation", fails for Verge(tor-4.3.5).

./configure --with-libraries=all 
--->> Compiles fine(unsure if complaints about "Deprecation", fails for Verge(tor-4.3.5).**_

For the above I have tried to compile with and without OpenSSL-1.1.1g.

So concluding factor for VirtualBox compilation is Verge(tor-4.3.5) compiles fantastically with Ubuntu-18.04 Boost-1.65 and with OpenSSL-1.1.1g. 
I don't know if VirtualBox is the issue for Boost-1.73.0.

Problem with my Ubuntu-18.04 bare-metal install is I had over written my Python3 with Python-3.8.3 and my Boost-1.65 with Boost-1.73.0.
I know it sounds weird. In any case, maybe this info could be of help on troubleshoot issues you or other may have.

I have the new Verge(tor-4.3.5) copied over to my bare-metal and seems to be running fine. I get peer connects and block updates.

Thanks for your input and help.
